x/auth/tx/aux_test.go

```diff
@@ -0,0 +1,147 @@
+package tx_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	clienttx "github.com/cosmos/cosmos-sdk/client/tx"
+	"github.com/cosmos/cosmos-sdk/simapp"
+	"github.com/cosmos/cosmos-sdk/testutil/testdata"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	txtypes "github.com/cosmos/cosmos-sdk/types/tx"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
+	authsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
+)
+
+// TestBuilderWithAux creates a tx with 2 aux signers:
+// - 1st one is tipper,
+// - 2nd one is just an aux signer.
+// Then it tests integrating the 2 AuxSignerData into a
+// client.TxBuilder created by the fee payer.
+func TestBuilderWithAux(t *testing.T) {
+	encCfg := simapp.MakeTestEncodingConfig()
+	testdata.RegisterInterfaces(encCfg.InterfaceRegistry)
+
+	// The final TX has 3 signers, in this order.
+	tipperPriv, tipperPk, tipperAddr := testdata.KeyTestPubAddr()
+	aux2Priv, aux2Pk, aux2Addr := testdata.KeyTestPubAddr()
+	feepayerPriv, feepayerPk, feepayerAddr := testdata.KeyTestPubAddr()
+
+	msg := testdata.NewTestMsg(tipperAddr, aux2Addr)
+	memo := "test-memo"
+	tip := &txtypes.Tip{Tipper: tipperAddr.String(), Amount: sdk.NewCoins(sdk.NewCoin("tip-denom", sdk.NewIntFromUint64(123)))}
+	chainID := "test-chain"
+	gas := testdata.NewTestGasLimit()
+	fee := testdata.NewTestFeeAmount()
+
+	// Create an AuxTxBuilder for tipper (1st signer)
+	tipperBuilder := clienttx.NewAuxTxBuilder()
+	tipperBuilder.SetAddress(tipperAddr.String())
+	tipperBuilder.SetAccountNumber(1)
+	tipperBuilder.SetSequence(2)
+	tipperBuilder.SetTimeoutHeight(3)
```

Comment:
Possibly a stupid question but what would happen if the tipper and the aux signer had different timeout heights? As far as I can tell this wouldn't throw any errors anywhere in this building process, and the final tx timeout will be whichever timeout is on the last signer data fed to `AddAuxSignerData()`. Actually I think this question extends to all the signer data that is not being set at a specific index in `AddAuxSignerData()`.

Reply:
not a stupid question! I thought about this too, but I went for the lazy way, which is that signature verification will fail when sending this tx (with 2 different timeout heights) to the node.

But maybe it's better to catch this early. I'll add some basic checks in AddAuxSignerData